client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-categories-order.ts

```diff
@@ -0,0 +1,45 @@
+import { useMemo } from 'react';
+import type { Category } from '../types';
+
+const patternCategoriesOrder = [
+	'featured',
+	'posts',
+	'text',
+	'gallery',
+	'images',
+	'call-to-action',
+	'banner',
+];
+
+const useCategoriesOrder = ( categories: Category[] ) => {
+	return useMemo(
```

Comment:
[nit] It might be slightly simple if we split `categories` into 2 arrays, and combine the result
* Categories with the pre-defined order
* The others